src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems

```diff
@@ -2783,4 +2783,7 @@
     <Compile Include="$(MSBuildThisFileDirectory)System\Numerics\IUnaryPlusOperators.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)System\Numerics\IUnsignedNumber.cs" />
   </ItemGroup>
+  <ItemGroup>
+    <AdditionalFiles Include="$(Version).versionstring" />
```

Comment:
Are there any side-effect from using the fake non-existent filename? For example, can this confuse incremental builds?

Reply:
I have no idea if fake filenames will have any effect, but .Combine() was affecting incremental generator's ability to cache. Just realized that source generators are also sharing the same $(Version) we are trying to obtain so I've replaced the file approach with InformationalVersion of srcgen assembly instead.